xo/src/move-maker.ts

```diff
@@ -41,16 +43,20 @@ export class MoveMaker {
     );
 
     let random = Math.floor(Math.random() * total) + 1; // Random inclusive between 1 and total
+    const length = counts.length;
     let move;
-    for (let i = 0; i < counts.length; i++) {
+
+    for (let i = 0; i < length; i++) {
       move = counts[i];
+
       if (move.count == 0) {
         continue;
       }
 
       if (random <= move.count) {
         return move;
       }
+
```

Comment:
You really like lots of white space

Reply:
ahhhh can't help it. I do it as I read through stuff

Reply:
and yes :)